src/wallet/wallet.cpp

```diff
@@ -1468,12 +1468,30 @@ void CWallet::transactionRemovedFromMempool(const CTransactionRef& tx, MemPoolRe
 
 void CWallet::blockConnected(ChainstateRole role, const interfaces::BlockInfo& block)
 {
-    if (role == ChainstateRole::BACKGROUND) {
-        return;
-    }
     assert(block.data);
     LOCK(cs_wallet);
 
+    switch (role) {
```

Comment:
be2be21d1740966325213258151d7f3477b19e1e: if #28608 lands first, this would check `role.validated` and `role.most_work` instead (cc @ryanofsky)